### ISSUE TITLE
gns3 2.2.8 and 2.2.9 vm grey (not green) under gns3 servers summary

### ISSUE THREAD

**Issue by jnatschev** [GNS3/gns3-gui#2999]
**Describe the bug**
the gns3 vm is grey (not green) under gns3 servers summary.
unable to determine where the bug is; that is, gns3, gns3 server, gns3 vm
gns3 and gns3 vm versions 2.2.5 to gns3 and gns3 vm 2.2.7 are stable and reliable

gns3 server:
server: local
host binding: ip address of vmnet1

gns3 vm:
virtualisation engine: vmware fusion pro 15.5.3
vm name: gns3 vm
ram: 12288MB
vcpus: 2

vmware fusion pro 15.5.3:
vm name: gns3 vm
processors: 2
memory: 12288MB
network adaptor: private to my mac
network adaptor 2: shared with my mac
compatibility: hardware version 16

**GNS3 version and operating system (please complete the following information):**
 - OS: MacOS Catalina 10.15.5
 - OS Firewall: on
 - GNS3 version 2.2.8
 - GNS3 VM version: 2.2.8
 - GNS3 version 2.2.9
 - GNS3 VM version: 2.2.9 

**To Reproduce**
---start gns3 2.2.5---
1. Install gns3 2.2.5
2. Install gns3 VM 2.2.5
3. Test (success: gns3 vm green under gns3 servers summary and able to add and use cisco csr routers to/in the topology)
---end gns3 2.2.5---

---start gns3 2.2.6---
1. Install gns3 2.2.6
2. Install gns3 VM 2.2.6
3. Test (success: gns3 vm green under gns3 servers summary and able to add and use cisco csr routers to/in the topology)
---end gns3 2.2.6---

---start gns3 2.2.7---
1. Install gns3 2.2.7
2. Install gns3 VM 2.2.7
3. Test (success: gns3 vm green under gns3 servers summary and able to add and use cisco csr routers to/in the topology)
---end gns3 2.2.7---

---start gns3 2.2.8---
1. Install gns3 2.2.8
2. Install gns3 VM 2.2.8
3. Test (fail: gns3 vm grey under gns3 servers summary and didn't even try to add and use cisco csr routers to/in the topology)
---end gns3 2.2.8---

---start gns3 2.2.9---
1. Install gns3 2.2.9
2. Install gns3 VM 2.2.9
3. Test (fail: gns3 vm grey under gns3 servers summary and didn't even try to add and use cisco csr routers to/in the topology)
---end gns3 2.2.9---

not sure of any other data that may be relevant.

**Comment by jnatschev** [comment]
oh. there is another thing i thought of.

with both gns3/gns3 vm 2.2.8 and gns3/gns3 vm 2.2.9, after both gns3 and gns3 vm had both started, i waited around five minutes and the gns3 vm under servers summary were still grey. hovering over the gns3 vm under servers summary displayed a small pop-up message with the words something like "discovering and connecting to ..."

**Comment by jnatschev** [comment]
ok. bit of an update. i noticed that 2.2.10 was released. i downloaded it (both gui and vm). installed it. started it all up and it didn't work. still perplexed i reinstalled version 2.2.7. started it all up and it worked. wondering what is going on i decided to try something different.

i realised that there must have been a change in workings from 2.2.7 to 2.2.8 and onwards.

so, i made a backup and removed the .config/GNS directory to enable a "fresh" install of 2.2.10.

installed the gns3 2.2.10 app and gns3 vm 2.2.10. started the gns3 vm 2.2.10. started the gns3 2.2.10 app. went through the initial setup.

it all worked.

so basically, it may be me not having read "install instructions" or it could still be a bug. that is a bug from the perspective that if there is an existing ".conf" file for 2.2 versions below 2.2.8 and a gns3 version >= 2.2.8 is installed the appropriate ".conf" file is not updated with the new configurations for versions >= 2.2.8.

so. with all that written, i can say that the connection between "server" and "vm" is working and all services under the bui "servers summary" are displayed as "green".

yipee!

**Comment by jnatschev** [comment]
i have tried a couple of projects and they all work fine.

2.2.10 seems pretty reliable/stable for me. obviously, needed to recreate dynamips images and qemu images. a bit of manual effort but all good.

let me know if you need any other sort of information.

love ya good work and keep up the good work.

**Comment by jtorri** [comment]
That gns3 2.2.10 is working consistently between reboots of workstation is great!  Would very much appreciate screenshots of the following from your configuration.
1. Server Preferences, both Main Server and Remote Servers
2. GNS3 VM Preferences
3. Detailed info of Windows versions, e.g., Windows 10 version 2004 OS build 19041.329
4. Whether you are running VMware Workstation Pro or VMware Player and its version details, e.g., VMware Workstation 15 Pro, 15.5.6 build-16341506 
That is a great deal of detail and work to ask but I've been running GNS3 since versions 1.x without show stopping issues.  However, since versions 2.2.7 and 2.2.8 have had significant problems sporadically largely related to connection to GNS3 VM.  My impression so far is they are largely related to updates by VMware and Windows.  No problems whatever running GNS3 on Linuxmint 19.2
And, yes, GNS3 is a remarkable piece of software!

**Comment by jnatschev** [comment]
@jtorri, here you go. not running windows. as in my orginial issue submission, i use macos. also, as in my original issue submission, running vmware fusion. if you haven't tried already you might take a backup of your gns3 configs, ensure gns3 is not running, then remove them. start gns3 so that it creates the configs from scratch. that's basically what i did.
![gns2-server-preferences](https://user-images.githubusercontent.com/8773561/85574702-58c82b00-b67a-11ea-8eee-24a3d47c37f1.png)
![gns3-remoteserver-preferences](https://user-images.githubusercontent.com/8773561/85574715-5bc31b80-b67a-11ea-8f5b-ce3ae3915589.png)
![gns3-vm-preferences](https://user-images.githubusercontent.com/8773561/85574721-5cf44880-b67a-11ea-8fc4-4d64d9f7b4b8.png)

**Comment by jtorri** [comment]
jnatschev, that was quick!  Thanks!
I believe the issue for me in Windows is the value for host binding in Server Preferences, Main Server.  Until recently a value of localhost has always worked.  As for recreating the conf files I have done that and agree it is a good place to start.
Thanks again!

**Comment by jnatschev** [comment]
i see what you're saying. there are number of ways you test this. on macos, i get a message if the "local server" and the "gns3 vm" are not associated to the same ip subnet. look at the configuration of your gns3 vm network adaptor; the network adaptor that is associated with a vmnet (on macos there is private to my mac, vmnet2, vmnet3, etc.). if you have the equivalent to macos "private to my mac" selected, then on your windows open a command-line terminal and execute "ipconfig /all". in the output look for the network adaptor vmnet1. for the network adaptor vmnet1 note the ip address. now in gns3 preferences -> server -> main server -> host binding, select from the drop-down list the ip address that relates to the ip address you noted for vmnet1.

my vmware fusion gns3 vm is setup as per the attached image. in vmware fusion, i created vmnet2, vmnet3, etc. i associated the gns3 vm network adapator to vmnet2. vmnet2 is associated with an ip subnet. my vmnet2 is associated with the ip subnet 192.168.255.0/24, ip address 192.168.255.1. so if you look at the image of my server -> main server -> host binding again you'll see it is associated with the 192.168.255.1 ip address.

from here i created two /etc/hosts entries:
192.168.255.1 gns
192.168.255.128 gnsvm

192.168.255.128 is the ip address of the gns3 vm.

now when i "telnet" to a network device in a topology and the device has been set to run on the gns3 vm i do the following:
telnet gnsvm port_number
when i "telnet" to a network device in a topology and the device has been set to run on the "local server i do the following:
telnet gns port_number

oh. i have added an image of the vmware fusion custom network adaptors (vmnet2, vmnet3, etc.).
![gns3-vm-preferences](https://user-images.githubusercontent.com/8773561/85637622-0a8d4900-b6c7-11ea-806d-d7f98ef3d463.png)
![vmwarefusion_preferences_network](https://user-images.githubusercontent.com/8773561/85637782-740d5780-b6c7-11ea-968f-c9668d390143.png)

**Comment by jtorri** [comment]
jnatschev
Wow!  That response took a good amount of thought and effort!  Many thanks!
While it seems any host-only vmnet will work I have always used vmnet1 for eth0 of the GNS3 VM.  Best I can tell there has been some inconsistency in vmnet1 obtaining a valid IP in my workstation's network adapter settings.  So I tried using vmnet2 which did obtain a valid workstation network adapter setting IP but still inconsistent connections to GNS3 VM.  Also, from my workstation I could ping the vmnet1 IP, e.g., 192.168.215.1, but not the GNS3 VM IP, e.g., 192.168.215.128.
Long story short I am guessing but it seems I had an older VIX versioin issue.  So, I completely uninstalled all VMware products, ran a registry cleaner, rebooted and installed VMware Workstation Pro 15.5.6 and continue to use GNS3 2.2.10
My GNS3 VM eth0 is set to vmnet1 and my GNS3 application Server Preferences, Main Server, Host Binding is now set to 192.168.174.1, the value of vmnet1 IP.  I believe this is a recommendation Jeremy made to a similar issue I posted a while back.
So, we'll see how it goes.  I'm not the sharpest tack in the box but I am stubborn and sometimes stubbornness and listening helps!
Anyway, thanks again for the help!

**Comment by jnatschev** [comment]
all good. happy gns3'ing!

**Comment by jtorri** [comment]
Based on an unbelievably inadequate amount of testing it appears the issue of vmnet1 issues on the VMware GNS3 VM are resolved with the release of GNS3 2.2.12.
I am using the GNS3 2.2.12 application and the VMware VM that come with GNS3 2.2.12 release.

GNS3 management console.
Running GNS3 version 2.2.12 on Windows (64-bit) with Python 3.6.8 Qt 5.12.1 and PyQt 5.12.
Copyright (c) 2006-2020 GNS3 Technologies.

I have noticed the following changes to the GNS3 application and to Windows 10.  
The following update was installed on my Windows 10 machine 8/12/2020
2020-08 Cumulative Update for Windows 10 Version 2004 for x64-based Systems (KB4566782)
resulting in new OS Build of Windows 10:  Windows 10 version 2004 (OS Build 19041.450)

I also noticed the GNS3 VM from GNS3 2.2.11 is running Ubuntu 18.04.4 LTS 
while the GNS3 VM from GNS3 2.2.12 is running Ubuntu 18.04.5 LTS

Obviously, the point being I am not certain what solved, or at least appears to have solved, the vmnet1 requiring disable/enable
to work and allow connection from the GNS3 server to the GNS3 VM.  Once the GNS3 VM has fully loaded the connection indicator on the GNS3 application turns green, no error message appears and it is possible to load and run a project.
